src/validation.cpp

```diff
@@ -3775,18 +3775,15 @@ bool CChainState::AcceptBlock(const std::shared_ptr<const CBlock>& pblock, Block
 bool ProcessNewBlock(const CChainParams& chainparams, const std::shared_ptr<const CBlock> pblock, bool fForceProcessing, bool *fNewBlock)
 {
     AssertLockNotHeld(cs_main);
-
     {
-        if (fNewBlock) *fNewBlock = false;
-        BlockValidationState state;
-
         LOCK(cs_main);
 
+        BlockValidationState state;
```

Comment:
nit: keep struct construction outside of cs_main, that's still a concurrency

Reply:
> that's still a concurrency

What you mean?

Reply:
I think @ariard means that there's no need to construct the `BlockValidationState` object within the `cs_main` scope. That's true, but constructing this object is very cheap, so I don't think it's a problem (and placing the variable declaration next to where it's used make this clearer).

I'm actually going to remove this commit from the PR, since I don't think it's necessary (and may make it more likely to introduce a bug, since the callers to `ProcessNewBlock()` in net_processing don't check the return code of the function before using the `fNewBlock` out param.